File: playwright/page_objects/text_box_page.py
from playwright.sync_api import Page
import allure

class TextBoxPage:
    # Initialize page and selectors
    def __init__(self, page: Page):
        self.page = page
        self.fullNameField = page.locator("#userName")
        self.emailField = page.locator("#userEmail")
        self.currentAddressField = page.locator("#currentAddress")
        self.permanentAddressField = page.locator("#permanentAddress")
        self.submitButton = page.locator("#submit")
        self.pageLoadingIndicator = page.locator(".text-center")


    # Actions on page
    def navigate(self, url: str):
        self.page.goto(url)


    # Get information from page
    def check_loading_page(self):
        return self.pageLoadingIndicator.is_visible()

    @allure.step("Fill Name field")
    def fill_name(self):
        return self.fullNameField.fill("Test_name")

    @allure.step("Click Submit button")
    def submit_input(self):
        return self.submitButton.click()

    @allure.step("Check submitted name")
    def filled_name_value(self,expected_name):
        return self.page.locator(f"//*[@id='name' and contains(., '{expected_name}')]")

File: js_playwright_demoqa_test/pages/textBoxPage.js
exports.TextBoxPage = class TextBoxPage {
  constructor(page) {
    this.page = page;
    this.fullNameInput = page.locator('#userName');
    this.emailInput = page.locator('#userEmail');
    this.currentAddressInput = page.locator('#currentAddress');
    this.permanentAddressInput = page.locator('#permanentAddress');
    this.submitButton = page.locator('#submit');
    this.outputName = page.locator('#output #name');
    this.outputEmail = page.locator('#output #email');
  }

  async navigate() {
    await this.page.goto('/text-box');
  }

  async fillForm({ name, email, currentAddress, permanentAddress }) {
    await this.fullNameInput.fill(name);
    await this.emailInput.fill(email);
    await this.currentAddressInput.fill(currentAddress);
    await this.permanentAddressInput.fill(permanentAddress);
    await this.submitButton.click();
  }

  async getOutput() {
    return {
      name: await this.outputName.textContent(),
      email: await this.outputEmail.textContent(),
    };
  }
};